geoalchemy2/admin/dialects/sqlite.py

```diff
@@ -172,29 +275,112 @@ def before_create(table, bind, **kw):
                     col.type, "spatial_index", False
                 ):
                     table.info["_after_create_indexes"].append(idx)
-    _setup_dummy_type(table, gis_cols)
+
+    if db_is_gpkg:
+        if len(gis_cols) > 1:
+            raise ValueError(
+                "Only one geometry column is allowed for a table stored in a GeoPackage."
+            )
+        elif len(gis_cols) == 1:
+            col = gis_cols[0]
+            srid = col.type.srid
+
+            if col.type.geometry_type is None:
+                col.type.geometry_type = "GEOMETRY"
+
+            # Add the SRID of the table in 'gpkg_spatial_ref_sys' if this table exists
+            if (
+                bind.execute(text("""PRAGMA main.table_info("gpkg_spatial_ref_sys");""")).fetchall()
```

Comment:
Why "if it exists"? A geopackage without this table is invalid. GDAL also raises if that is the case.

Reply:
Indeed, good point